### PR TITLE
Add missing deps in importer target

### PR DESCRIPTION
**Pull request by yrliou**
Fix https://github.com/brave/brave-browser/issues/6678

Add missing deps in importer target, `chrome/browser` is not added because of circular dependency, will create a follow up issue to break this and then importer target could include it as deps normally.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [x] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [x] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by mihaiplesa** [comment]
Unit tests reached timeout on Windows but looks good otherwise https://ci.brave.com/job/brave-browser-build-pr/job/fix_importer_deps/3/execution/node/734/log/

File: browser/importer/BUILD.gn [hunk 1]
```diff
@@ -22,7 +22,22 @@ source_set("importer") {
 
   deps = [
     "//base",
+    "//brave/common",
+    "//brave/common:pref_names",
+    "//brave/components/brave_rewards/browser",
+    "//chrome/browser/ui",
```
Comment (bridiver) [inline browser/importer/BUILD.gn]: ui? for importer? What uses this?

Comment (yrliou) [reply browser/importer/BUILD.gn]: In brave_profile_writer.cc,
```
#include "chrome/browser/ui/browser.h"
#include "chrome/browser/ui/browser_finder.h"
#include "chrome/browser/ui/browser_navigator.h"
#include "chrome/browser/ui/browser_navigator_params.h"
#include "chrome/browser/ui/browser_tabrestore.h"
#include "chrome/browser/ui/browser_window.h"
#include "chrome/browser/ui/tabs/tab_strip_model.h"
```

Comment (bsclifton) [reply browser/importer/BUILD.gn]: I believe this was added when writing the code to import open windows/tabs from Muon:
https://github.com/brave/brave-core/pull/981

**Comment by yrliou** [comment]
Weird that unit-test on Windows keep abort without real error, I'm going to apply skip-CI labels on other platforms to retry Windows again.

**Comment by mihaiplesa** [comment]
Think it's timing out. It has a 20m timeout set at https://github.com/brave/brave-browser/blob/master/Jenkinsfile#L730. We can increase if need be.

**Comment by yrliou** [comment]
> Think it's timing out. It has a 20m timeout set at https://github.com/brave/brave-browser/blob/master/Jenkinsfile#L730. We can increase if need be.
Thanks for pointing that out @mihaiplesa, Windows is green on my third try here! :)

**Comment by yrliou** [comment]
Create a follow up issue at https://github.com/brave/brave-browser/issues/6726 for a few pieces that are still missing.

**Pull request merged**